proof/courier.go

```diff
@@ -30,9 +32,48 @@ const (
 
 	// ApertureCourier is a courier that uses the hashmail protocol to
 	// deliver proofs.
+	//
+	// TODO(ffranr): Rename to HashmailCourier (use protocol name rather
+	//  than service).
 	ApertureCourier
 )
 
+// Convert CourierProtocol to string
+func (p CourierType) String() string {
+	switch p {
+	case ApertureCourier:
+		return "hashmail"
+	default:
+		return "disabled_courier"
+	}
+}
+
+// CourierAddr is the address of the proof courier service that will be used to
+// transfer proofs related to the asset send process.
+type CourierAddr struct {
+	// Protocol is the protocol of the proof courier.
+	Protocol CourierType `long:"protocol" choice:"hashmail" description:"Proof courier service protocol."`
```

Comment:
This is meant to be a global value/setting? I would think instead we have the initial config, then the user can override that whenever they make an address.

Reply:
This effectively allows us to set a default courier address when starting the daemon. Then, if a courier address isn't specified when generating a tap address via RPC, we can fall back to this default courier address.